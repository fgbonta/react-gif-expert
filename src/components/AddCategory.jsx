//import React from 'react'
import { useState } from "react";
import PropTypes from 'prop-types';

//export const AddCategory = ({ setCategories }) => {
export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState("");

    const onInputChange = ( { target }) => {        
        setInputValue(target.value);
    }

    const onFormSubmit = ( event ) => {
        //console.log('hola form');
        event.preventDefault();
        
        if(inputValue.trim().length <= 1) return;

        //le mandas a la fn un callback
        //setCategories( categories => [inputValue.trim(),...categories] );
        onNewCategory(inputValue.trim());
        
        //limpiamos el input
        setInputValue("");
    }

    return (
        <form onSubmit={ (event) => onFormSubmit(event) } aria-label="form-add">
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ (event) => onInputChange(event) }
            />
        </form>        
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
