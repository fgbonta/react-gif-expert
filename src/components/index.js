
//re-export
export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';