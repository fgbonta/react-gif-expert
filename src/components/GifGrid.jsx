import PropTypes from 'prop-types';
import { GifItem } from '../components/GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({ category }) => {  

    /**
     * cuando hay un cambio en el state
     * react redibuja el componente
     */
    const { images, isLoading } = useFetchGifs(category);    

    return (
        <>
            <h3>{ category }</h3>
            {
                /* isLoading && (<h2>cargando..</h2>) */
                isLoading
                ? (<h2>cargando..</h2>)
                : null
                /* el null no se renderiza */
            }
            <div className="card-grid">
                {
                    images.map( ( image ) => (
                        <GifItem 
                            key={ image.id }
                            { ...image }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}