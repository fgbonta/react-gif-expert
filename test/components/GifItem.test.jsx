import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => { 

    const title = 'One Punch Man GIF';
    const url = 'https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=f65293704ra1156a44w6mn97vnbpxdui8ndy3pvkvp1gnyj2&rid=giphy.gif&ct=g';
        
    test('Debe hacer match con el snapshot', () => { 

        const { container } = render(<GifItem title={title} url={url} />);        

        expect( container ).toMatchSnapshot();
        
     });

     test('Debe de mostrar la img con el URL y el ALT indicado', () => { 

        render(<GifItem title={title} url={url} />);
        //screen.debug();

        const { src , alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title); 
        
     });

     test('Debe de mostrar el titulo en el componente', () => { 

        const {container } = render(<GifItem title={title} url={url} />);                
        //screen.debug();

        /**
         * son 2 formas de evaluar
         * la segunda es mas precisa
         */
        expect( screen.getByText(title) ).toBeTruthy();
        expect( container.querySelector('div.card > p').innerHTML ).toBe(title); 
        
     })
 });