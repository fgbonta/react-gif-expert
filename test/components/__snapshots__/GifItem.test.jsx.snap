// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man GIF"
      src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=f65293704ra1156a44w6mn97vnbpxdui8ndy3pvkvp1gnyj2&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man GIF
    </p>
  </div>
</div>
`;
